crates/codegen/parser/runtime/src/napi_interface/cursor.rs

```diff
@@ -53,8 +52,8 @@ impl Cursor {
     }
 
     #[napi(ts_return_type = "cst.Node", catch_unwind)]
-    pub fn node(&self, env: Env) -> JsObject {
-        self.0.node().to_js(env)
+    pub fn node(&self) -> Either<RuleNode, TokenNode> {
```

Comment:
It's worth noting that we can't use Rust type alias here (or anywhere else below), i.e. `type TsNode = Either<RuleNode, TokenNode>`.

Because napi-rs does not keep a database and only visits what is attributed explicitly with the `#[napi]` macro (which also doesn't support type aliases), it cannot see or understand that a theroretical `-> TsNode` in fact means `Either<...>`; it just type-gens a `TsNode` return type, which fails later at TS compile-time, as a `TsNode` type is never defined in TS.